src/main/java/uk/org/tombolo/execution/DataExportEngine.java

```diff
@@ -101,4 +103,38 @@ private void importDatasource(ImporterMatcher forceImports, DatasourceRecipe dat
 				forceImports.doesMatch(datasourceSpec.getImporterClass())
 		);
 	}
+
+	private Importer initialiseImporter(String importerClass, String configFile) throws Exception {
+		Config importerConfiguration = null;
+		if (configFile != null && !"".equals(configFile)) {
+			importerConfiguration = ConfigUtils.loadConfig(
+					AbstractRunner.loadProperties("Configuration file", configFile));
+
+		}
+		return (Importer) Class.forName(importerClass).getDeclaredConstructor(Config.class).newInstance(importerConfiguration);
+	}
+
+	private String verifyProvider() throws Exception {
+		String validProvider = null;
+
+		File specificationFile = new File(DataExportRunner.executionSpecPath);
+		if (!specificationFile.exists()) return validProvider;
+		JSONReader reader = new JSONReader(specificationFile, new ArrayList<>(Arrays.asList("importerClass", "provider")));
+		reader.getData();
+		ArrayList<String> importers = reader.getTagValueFromAllSections("importerClass");
```

Comment:
Would this be still valid with the changes like in #296? In this case we could have the provider of a subject implicitly imported inside other importers.

Reply:
Yes it checks for all the providers, irrespective of the level at which they are provided in the recipe, all the providers are checked in a recipe for their validity

Reply:
This will miss the modelling fields providers, right?

Reply:
Yeah it will miss the modelling fields, it is only for main recipe